lapack/testlapack/dlangb.go

```diff
@@ -0,0 +1,109 @@
+// Copyright ©2021 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package testlapack
+
+import (
+	"fmt"
+	"math"
+	"testing"
+
+	"golang.org/x/exp/rand"
+
+	"gonum.org/v1/gonum/floats"
+	"gonum.org/v1/gonum/lapack"
+)
+
+type Dlangber interface {
+	Dlangb(norm lapack.MatrixNorm, m, n, kl, ku int, ab []float64, ldab int) float64
+}
+
+func DlangbTest(t *testing.T, impl Dlangber) {
+	rnd := rand.New(rand.NewSource(1))
+	for _, norm := range []lapack.MatrixNorm{lapack.MaxAbs, lapack.MaxRowSum, lapack.MaxColumnSum, lapack.Frobenius} {
+		t.Run(normToString(norm), func(t *testing.T) {
+			for _, n := range []int{0, 1, 2, 3, 4, 5, 10} {
+				for _, m := range []int{0, 1, 2, 3, 4, 5, 10} {
+					for _, kl := range []int{0, 1, 2, 3, 4, 5, 10} {
+						for _, ku := range []int{0, 1, 2, 3, 4, 5, 10} {
+							for _, ldab := range []int{kl + ku + 1, kl + ku + 1 + 7} {
```

Comment:
I removed the `iter` loop here because it increases the test run time disproportionately and the NaN element will be inserted in some cases anyway.